internal/integrationtests/fixtures.go

```diff
@@ -7,73 +7,366 @@ import (
+// &txnbuild.PathPaymentStrictSend{},
+// &txnbuild.ManageBuyOffer{},
+// &txnbuild.SetOptions{},
+// &txnbuild.Clawback{},
+// &txnbuild.SetTrustLineFlags{},
+// --- Soroban:
+// &txnbuild.InvokeHostFunction{},
+
+// Missing
+// --- Classic:
+// &txnbuild.ClawbackClaimableBalance{},
+// &txnbuild.LiquidityPoolDeposit{},
+// &txnbuild.LiquidityPoolWithdraw{},
+// &txnbuild.CreateClaimableBalance{},
+// &txnbuild.ClaimClaimableBalance{},
+// &txnbuild.RevokeSponsorship{},
+// --- Soroban:
+// &txnbuild.ExtendFootprintTtl{},
+// &txnbuild.RestoreFootprint{},
+
 type Fixtures struct {
-	NetworkPassphrase string
-	SourceAccountKP   *keypair.Full
-	RPCService        services.RPCService
+	NetworkPassphrase  string
+	PrimaryAccountKP   *keypair.Full
+	SecondaryAccountKP *keypair.Full
+	RPCService         services.RPCService
 }
 
-// prepareClassicOps prepares a slice of strings, each representing a payment operation XDR. Currently only returns one operation (payment).
-func (f *Fixtures) prepareClassicOps() ([]string, error) {
+// preparePaymentOp creates a payment operation.
+func (f *Fixtures) preparePaymentOp() (string, *Set[*keypair.Full], error) {
 	paymentOp := &txnbuild.Payment{
-		SourceAccount: f.SourceAccountKP.Address(),
-		Destination:   f.SourceAccountKP.Address(),
+		SourceAccount: f.PrimaryAccountKP.Address(),
+		Destination:   f.PrimaryAccountKP.Address(),
 		Amount:        "10",
 		Asset:         txnbuild.NativeAsset{},
 	}
 
 	paymentOpXDR, err := paymentOp.BuildXDR()
 	if err != nil {
-		return nil, fmt.Errorf("building payment operation XDR: %w", err)
+		return "", nil, fmt.Errorf("building payment operation XDR: %w", err)
 	}
 	b64OpXDR, err := utils.OperationXDRToBase64(paymentOpXDR)
 	if err != nil {
-		return nil, fmt.Errorf("encoding payment operation XDR to base64: %w", err)
+		return "", nil, fmt.Errorf("encoding payment operation XDR to base64: %w", err)
+	}
+
+	return b64OpXDR, NewSet(f.PrimaryAccountKP), nil
+}
+
+// prepareSponsoredAccountCreationOps creates an account using sponsored reserves.
+// NOTE: the account created here is meant to be deleted at a later time through an account merge operation.
+// NOTE 2: one manageData operation is included here as a bonus.
+func (f *Fixtures) prepareSponsoredAccountCreationOps(newAccountKP *keypair.Full) ([]string, *Set[*keypair.Full], error) {
+	operations := []txnbuild.Operation{
+		&txnbuild.BeginSponsoringFutureReserves{
+			SponsoredID:   newAccountKP.Address(),
+			SourceAccount: f.PrimaryAccountKP.Address(),
+		},
+		&txnbuild.CreateAccount{
+			Amount:        "0",
+			Destination:   newAccountKP.Address(),
+			SourceAccount: f.PrimaryAccountKP.Address(),
+		},
+		&txnbuild.ManageData{
+			Name:          "foo",
+			Value:         []byte("bar"),
+			SourceAccount: f.PrimaryAccountKP.Address(),
+		},
+		&txnbuild.EndSponsoringFutureReserves{
+			SourceAccount: newAccountKP.Address(),
+		},
+	}
+
+	b64OpsXDRs := make([]string, len(operations))
```

Comment:
Code quality nit: The code in lines 112-124 are repeated across the functions. We can abstract them into a separate function instead

Reply:
https://github.com/stellar/wallet-backend/pull/167/commits/2f5a3f2280cb9ec289a2224a57cb18aee64d1148